golang/defaults.yaml

```diff
@@ -12,15 +12,15 @@ golang:
     name: golang
     use_upstream_repo: false
     use_upstream_archive: true
+    uri: https://storage.googleapis.com/golang
```

Comment:
@myii  This was the fix for "_TypeError: unzip() got an unexpected keyword argument 'uri'" ... moving "uri_" up one level.  Not sure why your GitHub Actions is still complaining.

Reply:
@noelmcloughlin Well, the `default` (archive) suite is still using `pillar.example` as its pillar:

https://github.com/saltstack-formulas/golang-formula/blob/43b3b932a22dc2bf26e1f23642741d7c6d2931ce/kitchen.yml#L322-L323

And that still has the `uri` defined:

https://github.com/saltstack-formulas/golang-formula/blob/master/pillar.example#L11-L13

I've just tried by commenting that out and the states have passed.  So now need to figure out what the right situation is for _all_ platforms.

Reply:
@noelmcloughlin So `uri` and `archive_suffix` only appear to be used for `golang.pkg.use_upstream_archive`:

https://github.com/saltstack-formulas/golang-formula/blob/32e1fc77d4aa351a49f01b335402c665280828f9/golang/map.jinja#L57-L65

So shouldn't all of these occurrences be put underneath that key?

Reply:
This is the problem behind everything:

https://github.com/saltstack-formulas/golang-formula/blob/32e1fc77d4aa351a49f01b335402c665280828f9/golang/archive/install.sls#L25-L27

The Windows `archive.extracted` state doesn't handle these values, so this whole section shouldn't be done in this way.  The implementation is messy, so something has to give.  Really, nothing should be under `golang.pkg.archive` except that it appears in the docs under https://docs.saltproject.io/en/latest/ref/states/all/salt.states.archive.html.

Reply:
Alright, I've placed those non-state entries under a new `archive_helper` key and that seems to be a lot cleaner.  Testing on all platforms again.

Reply:
@myii good idea, I like `archive_helper`.